src/server_manager/web_app/digitalocean_account.ts

```diff
@@ -47,14 +47,26 @@ export class DigitalOceanAccount implements digitalocean.Account {
   }
 
   async getStatus(): Promise<digitalocean.Status> {
-    const account = await this.digitalOcean.getAccount();
-    if (account.status === 'active') {
-      return digitalocean.Status.ACTIVE;
-    }
-    if (!account.email_verified) {
-      return digitalocean.Status.EMAIL_UNVERIFIED;
-    }
-    return digitalocean.Status.MISSING_BILLING_INFORMATION;
+    const [account, droplets] = await Promise.all([
+      this.digitalOcean.getAccount(),
+      this.digitalOcean.getDroplets(),
+    ]);
+    const needsEmailVerification = !account.email_verified;
+    // If the account is locked for no discernible reason, and there are no droplets,
+    // assume the billing info is missing.
+    const needsBillingInfo =
+      account.status === 'locked' && !needsEmailVerification && droplets.length == 0;
+    const hasReachedLimit = droplets.length >= account.droplet_limit;
+    const warning = account.status !== 'active' ? account.status_message : '';
+    const detail = `status: "${account.status}"`;
```

Comment:
Don't format the detail in the model. That's an app concern.
I wouldn't mind adding the "(status: ..)" to the warning though.

Reply:
OK, done